src/passes/Heap2Local.cpp

```diff
@@ -688,6 +707,27 @@ struct Struct2Local : PostWalker<Struct2Local> {
     replaceCurrent(builder.makeBlock(contents));
   }
 
+  void visitRefEq(RefEq* curr) {
+    if (!analyzer.reached.count(curr)) {
+      return;
+    }
```

Comment:
Just to make sure I understand: we've done a data flow analysis and determined that this expression is not dead? Is it possible for `reached` to be nonzero and still have type `Type::unreachable` below?

Reply:
Ah, yeah, "reach" is used in two ways here. `reached` means that the allocation, the `struct.new / array.new`, reaches this location. So this first check is just making sure that we only look in the places we traced the path of the allocation.

Reply:
Consider
```wat
(ref.eq
  (unreachable)
  (struct.new ..)
)
```
That allocation reaches the `ref.eq`, but also the `ref.eq` is unreachable code, separately.